transformer_engine/common/gemm/cublaslt_gemm.cu

```diff
@@ -325,6 +325,36 @@ void cublas_gemm(const Tensor *inputA, const Tensor *inputB, Tensor *outputD,
   const bool use_fp8 = is_fp8_dtype(param.Atype) || is_fp8_dtype(param.Btype);
   const bool use_fp4 = is_fp4_dtype(param.Atype) || is_fp4_dtype(param.Btype);
 
+  // Tensor scaling factors for NVFP4
+  if (use_fp4 && inputA->amax.dptr && inputB->amax.dptr) {
+    // Include tensor scales in alpha for NVFP4 GEMMs when amaxs are provided
+    // Note: Store alpha in user-provided workspace and store beta in the device constant memory.
+    NVTE_CHECK(workspaceSize >= 4, "NVFP4 GEMM requires at least 4 byte workspace, but got ",
+               workspaceSize, " bytes.");
+    workspaceSize = (workspaceSize / 4) * 4 - 4;  // Align to 4 bytes and remove last 4 bytes
+    uint8_t *workspace_ptr = reinterpret_cast<uint8_t *>(workspace);
+    float *alpha_ptr = reinterpret_cast<float *>(alpha);
+    float *D_tensor_scale_inv_ptr = reinterpret_cast<float *>(&workspace_ptr[workspaceSize]);
+    TensorWrapper D_tensor_scale_inv(D_tensor_scale_inv_ptr, std::vector<size_t>{1},
+                                     DType::kFloat32);
```

Comment:
Nit: `D_tensor_scale_inv` is not a good name because it doesn't have anything to do with the D tensor or scale-inv (at best is it the encoding scale for D, but only when beta=0 and D is unquantized).
```suggestion
    float *updated_amax_ptr = reinterpret_cast<float *>(&workspace_ptr[workspaceSize]);
    TensorWrapper updated_amax(updated_amax_ptr, std::vector<size_t>{1}, DType::kFloat32);
```